src/plugins/explore/public/application/register_tabs.ts

```diff
@@ -44,18 +50,32 @@ export const registerBuiltInTabs = (tabRegistry: TabRegistryService) => {
     order: 15,
     supportedLanguages: [EXPLORE_DEFAULT_LANGUAGE],
 
-    prepareQuery: (query) => {
-      return typeof query.query === 'string' ? query.query : '';
+    prepareQuery: (query, state) => {
```

Comment:
is this the only place we are using the prepareQuery? if not how come we don't just prepare the query from the patterns? why does a higher component need to handle this?

Reply:
this here is defining prepareQuery from the tab registry's definition, there is other code that calls upon pattern tab's prepareQuery for things such as figuring out what the cache key would be, sending out the query, etc.

Reply:
prepareQuery is used by all tabs that needs a custom prepare query